### DISCUSSION TITLE
Ease overwrite anxiety by marking shadows as changed/unchanged

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#5513] [Ideas]
*"Warning, don't edit shadow tiddlers because..."*

During imports or attempted upgrades we are alerted about conflicting version numbers. Presumably there's some little mechanism comparing these numbers. 

For shadows, could this same mechanism (or some other for that matter) be used to compare all the fields of the updated shadow with the original shadow to inform if there is actually any difference between these? This is highly valuable info for *overwritten shadow tiddlers*. I'm not asking for a full text diff comparison (alto that would be nice), just an indication if the updated shadow is identical or not to the old one - (bare the date which is likely changed for the update). Often individual shadow tids are not updated in an updated plugin.

Overwrites might not be as problematic if there was an indication that "compared to the last version, this shadow tiddler has changed / not changed".

**Comment by pmario** [comment]
The import handling already has a full fetched diffing mechanism. You can diff fields or the full-text if you want to. ... Is that not enough?

**Reply by twMat** [reply]
>The import handling already has a full fetched diffing mechanism. You can diff fields or the full-text if you want to. 

OK, so I've missed this... which then of itself is an issue, i.e how is this accessed or presented? But great to hear that there is one!

>... Is that not enough?

As I just posted elsewhere, the problem is this:

- How do you know if the upgrade has changed the shadow; often individual shadow tiddlers in an updated plugin remain unchanged.
- ...so if the upgrade didn't change the shadow then you just want to keep your overwrite. (Maybe you don't even need to be alerted about it.)
- ...but if the upgrade did change the shadow, then you need to be alerted so you can investigate what to do with your overwrite.

**Reply by pmario** [reply]
I think that's a sensible request.

**Comment by twMat** [comment]
So, here is my proposal.

I wish for the import mechansim to do the following:

1. Check (a) which shadows´have overwrites in the plugin and (b) which of these shadows that are modified in the update.
2. If yes to both (a) and (b), then set a "updated-shadowtiddler:yes" field in the overwrite.

I can imagine that this would not be a fool proof method to find what needs attention but it is definitely a good indicator, and it would substantially simplify handling of overwrites. The import mechanism is in JS so it is beyond a regular TW user to implement this proposal. But if this is in place, it is simple enough to build further things from there, in wikitext, like alert-lists of overwrites that need attention after an update, etc.

Anyone wanting to add something before I make an official request for this?

**Comment by twMat** [comment]
Posted Issue: [Indicated overwrites that need attention from updates](https://github.com/Jermolene/TiddlyWiki5/issues/5518)

Give it a thumbs up there if you agree this would be useful.
